### PR TITLE
uORB: locking fixes and status output

### PR DESCRIPTION
**Pull request by bkueng**
@julianoes can you test this on Snappy?

Replaces  #4758

### CODE REVIEW

File: src/modules/uORB/uORBDevices_nuttx.hpp [hunk 1]
```diff
@@ -222,6 +228,10 @@ class uORB::DeviceNode : public device::CDev
 	bool    _IsRemoteSubscriberPresent;
 	int32_t _subscriber_count;
 
+	//statistics
+	uint32_t _lost_messages = 0; ///< nr of lost messages for all subscribers. If two subscribers lose the same
+	///message, it is counted as two.
```
Comment (julianoes) [inline src/modules/uORB/uORBDevices_nuttx.hpp]: Good to know :).

**Comment by julianoes** [comment]
Bench tested on Snappy, everything is as before.

**Comment by LorenzMeier** [comment]
Going to merge this after the 1.4 release, which just depends on estimator tweaks now.

**Comment by LorenzMeier** [comment]
Rebased and applied.

**Pull request closed without merging**